### PR TITLE
GH-3034: IQ API's for remote state stores

### PR DESCRIPTION
**Pull request by sobychacko**
Fixes: #3034

https://github.com/spring-projects/spring-kafka/issues/3034

* Kafka Streams interactive query API's for accessing remote state stores.

### CODE REVIEW

File: spring-kafka/src/main/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryService.java [hunk 1]
```diff
@@ -75,20 +81,79 @@ public void setRetryTemplate(RetryTemplate retryTemplate) {
...
+			try {
+				return this.kafkaStreams.store(storeQueryParams);
+			}
+			catch (Exception e) {
+				throw new IllegalStateException("Error retrieving state store: " + storeName, e);
+			}
+		});
+	}
+
+	private void primeKafkaStreamsObject() {
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryService.java:97]: `setupKafkaStreams()` or `populate`?
I don't see a reason in `Object` suffix at all.
And never saw `prime` as a verb 😃

File: spring-kafka/src/main/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryService.java [hunk 2]
```diff
@@ -75,20 +81,79 @@ public void setRetryTemplate(RetryTemplate retryTemplate) {
...
+
+	/**
+	 * Gets the current {@link HostInfo} where this Kafka Streams application is running on.
+	 * This {link @HostInfo} is different from the Kafka `bootstrap.server` property, and is based on
+	 * the Kafka Streams configuration property `application.server` where user-defined REST
+	 * endpoints can be invoked per each Kafka Streams application instance.
+	 * If this property - `application.server` - is not available from the end-user application, then null is returned.
+	 * @return the current {@link HostInfo}
+	 */
+	public HostInfo getCurrentKafkaStreamsApplicationHostInfo() {
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryService.java:113]: `@Nullable`

File: spring-kafka/src/main/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryService.java [hunk 3]
```diff
@@ -75,20 +81,79 @@ public void setRetryTemplate(RetryTemplate retryTemplate) {
...
 		if (this.kafkaStreams == null) {
 			this.kafkaStreams = this.streamsBuilderFactoryBean.getKafkaStreams();
 		}
 		Assert.notNull(this.kafkaStreams, "KafkaStreams cannot be null. " +
 				"Make sure that the corresponding StreamsBuilderFactoryBean has started properly.");
-		StoreQueryParameters<T> storeQueryParams = StoreQueryParameters.fromNameAndType(storeName, storeType);
+	}
+
+	/**
+	 * Gets the current {@link HostInfo} where this Kafka Streams application is running on.
```
Comment (artembilan) [inline spring-kafka/src/main/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryService.java:106]: Method Javadocs must be imperative.
Kinda we give a command to do from the first verb.

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-kafka/src/main/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryService.java (hunk 1 above)
Comment (sobychacko) [reply spring-kafka/src/main/java/org/springframework/kafka/streams/KafkaStreamsInteractiveQueryService.java:97]: Prime can be a verb, but I get your point:)

**Review by artembilan — APPROVED** [review]
LGTM.

But we cannot merge for now since GH is out of order right now: https://www.githubstatus.com/

**Pull request merged**